lib/k8s/stack.rb

```diff
@@ -129,7 +132,10 @@ def keep_resource!(resource)
     # @param resource [K8s::Resource]
     # @return [Boolean]
     def keep_resource?(resource)
-      @keep_resources["#{resource.kind}:#{resource.metadata.name}@#{resource.metadata.namespace}"] == resource.metadata&.annotations&.dig(@checksum_annotation)
+      keep_annotation = @keep_resources["#{resource.kind}:#{resource.metadata.name}@#{resource.metadata.namespace}"]
+      return false unless keep_annotation
+
```

Comment:
I think we should check the owner reference count here before trying to check anything else. if there's owners, keep it.

Reply:
the` keep_resource?` won't be called when pruning with `keep_resources: false` (`stack.delete` / `K8s::Stack.delete`)